### DISCUSSION TITLE
Environment setup help

### DISCUSSION THREAD

**Discussion by HxriM** [microsoft/vscode-cmake-tools#3378] [General]
Hi!
I am developing in a Devcontainer and I need to source a complex environment file to get my toolchain set up properly.
Whats the best way to accomplish this? I can add it to the bashrc or /etc/profile but those files dont seem to be used by the CMake process. When starting CMake is it possible to instruct the Extension to use a login/interactive shell so these files can be accessed?

**Comment by gcampbell-msft** [comment]
Have you considered using CMakePresets.json, where you can explicitly set your toolchain file as well as setting up your environment in the `environment` field of a Configure Preset?
